contrib/pkg/awstagdeprovision/awstagdeprovision.go

```diff
@@ -266,6 +267,106 @@ func deleteLBs(vpc *ec2.Vpc, awsSession *session.Session, logger log.FieldLogger
 	return false
 }
 
+// filterV2LBsByVPC will find all the load balancers in the provided list that are under the provided VPC
+func filterV2LBsByVPC(lbs []*elbv2.LoadBalancer, vpc *ec2.Vpc, logger log.FieldLogger) []*elbv2.LoadBalancer {
+	filteredLBs := []*elbv2.LoadBalancer{}
+
+	for _, lb := range lbs {
+		if *lb.VpcId == *vpc.VpcId {
+			filteredLBs = append(filteredLBs, lb)
+		}
+	}
+
+	return filteredLBs
+}
+
+func deleteV2LBs(vpc *ec2.Vpc, awsSession *session.Session, logger log.FieldLogger) bool {
+	logger.Debugf("Deleting V2 load balancers (%s)", *vpc.VpcId)
+	defer logger.Debugf("Exiting deleting V2 load balancers (%s)", *vpc.VpcId)
+	elbv2Client := elbv2.New(awsSession)
+
+	total := 0
+	filteredLBs := []*elbv2.LoadBalancer{}
+	if err := elbv2Client.DescribeLoadBalancersPages(&elbv2.DescribeLoadBalancersInput{}, func(results *elbv2.DescribeLoadBalancersOutput, lastPage bool) bool {
+		total += len(results.LoadBalancers)
+		filteredLBs = append(filteredLBs, filterV2LBsByVPC(results.LoadBalancers, vpc, logger)...)
+		return lastPage
+	}); err != nil {
+		logger.Errorf("Error listing V2 load balancers: %v", err)
+		return false
+	}
+	logger.Debugf("from %d total V2 load balancers, %d scheduled for deletion", total, len(filteredLBs))
+
+	if len(filteredLBs) == 0 {
+		// no items left to delete
+		// see if we can delete target groups.
+		return deleteTargetGroups(vpc, awsSession, logger)
+	}
+
+	for _, lb := range filteredLBs {
+		logger.Debugf("Deleting V2 load balancer: %v", *lb.LoadBalancerName)
+		_, err := elbv2Client.DeleteLoadBalancer(&elbv2.DeleteLoadBalancerInput{
+			LoadBalancerArn: lb.LoadBalancerArn,
+		})
+		if err != nil {
+			logger.Debugf("Error deleting V2 load balancer %v: %v", *lb.LoadBalancerName, err)
+		} else {
+			logger.WithField("name", *lb.LoadBalancerName).Info("Deleted load balancer")
```

Comment:
You're probably just following an existing pattern, but it's a bit strange to me that we use `WithField` for name here but `: %v` for the name a few lines up in `Deleting V2 load balancer: %v`.  Should we use `WithField` there too?  If so, do we want to do that in this PR for your additions, or punt it to a whole-file-fix later PR?

Reply:
punt it to a whole-file-fix later PR is what I want :)